src/hooks/usePaymentsData.ts

```diff
@@ -0,0 +1,30 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import { useEffect } from 'react';
+import { useRecoilValue } from 'recoil';
+import { USER } from '../constants';
+import { $CartList, $CheckedCartIdList } from '../recoil/atom';
+import useGetQuery from './useGetQuery';
+import type { PaymentsData } from '../types';
+
+const usePaymentsData = (currentServerUrl: string) => {
+  const cartList = useRecoilValue($CartList(currentServerUrl));
+  const checkedCartIdList = useRecoilValue($CheckedCartIdList(currentServerUrl));
+
+  const queryParams = new URLSearchParams();
+  checkedCartIdList.forEach(id => {
+    queryParams.append('cartItemIds', String(id));
+  });
```

Comment:
이 코드의 연산이 꽤 될 수 있다고 생각되는데, useMemo를 추천드립니다! `queryParams.toString()`까지 작업해두면 어떨까요?

```suggestion
  const queryParams = useMemo(() => {
    const params = new URLSearchParams();
    checkedCartIdList.forEach(id => {
      params.append('cartItemIds', String(id));
    });
    return params.toString();
  }, [checkedCartIdList]);
```

Reply:
오..! 감사합니다! 반영해 놓겠습니다~!

마침 메모이제이션 관련 피드백이 와서 그런데 물어보고 싶은 점이 있습니다!
파랑의 리뷰를 굉장히 많이 봤었는데(이건 리뷰어들 리뷰를 모두 취합하는 스터디를 하고 있어서 그렇습니다..ㅎㅎ), 정말 메모이제이션을 사랑하시더라구요..! 그만큼 잘 아신다고 생각해서 아직 답을 내리지 못한 제 궁금증을 여쭈어 보고 싶어요!

지금 리뷰가 `useMemo` 이니 `useMemo` 를 예시로 들겠습니다!
사실 저는 지금까지 `useMemo` 가 효율적인 역할을 하기 위해서는 정말 연산이 깊어야된다고 생각하고 있었습니다!
아니면, 연산이 렌더링에 영향을 많이 주는 경우에 사용해 불필요한 재렌더링을 방지할 수 있을 것 같습니다!

이 연산 복잡도의 기준을 잡기 참 어려운 것 같습니다.. 환경에 따라 달라질 수 있는 부분이기도 하고, 제 스스로 이 임계값을 정하다는 것이 아직 잘 와닿지 않습니다. 사실 이번 미션에 메모이제이션을 거의 적용하지 않은 것도, 애플리케이션의 규모가 작다고 느꼈기 때문에 필요성을 못 느꼈기 때문입니다.

그래서 메모이제이션을 적용할 때 파랑만의 기준이 있는지 궁금합니다. 아니면, 회사에서 개발자들끼리 '어느 정도의 복잡도에서 메모이제이션을 정하자.' 라는 것이 존재하는지도 궁금합니다..!

Reply:
오히려 규모가 작았을 때 렌더링 비용을 `공부`해둬야 한다는 취지에서 메모이제이션을 꼭꼭 이야기 하고 있습니다.
생각보다 react hook을 다양하게 사용하지 않음으로 인해 일어나는 프로덕션에서의 문제가 많아요.
메모이제이션을 조금만 활용해도 성능이 크게 개선되거나, ref 나 LayoutEffect 등 대표적으로 아는 훅들도 도움이 많이 돼죠.

현재 구현한 내용으로 볼 때는 메모이제이션만 해당 되는 이야기라 지속적으로 언급할 뿐입니다😊
연산 비용에 대해서는 원시값을 핸들링하지 않고 객체를 다루거나 O(n) 이상 되는 시간/공간 복잡도를 가졌다면 메모이제이션 해두라 코멘트를 하고 있어요.